torch/distributed/fsdp/_optim_utils.py

```diff
@@ -1220,6 +1238,9 @@ def _optim_state_dict(
         if using_optim_input
         else _get_param_id_to_param(optim)
     )
+    fqn_to_fsdp_param_info = _get_fqn_to_fsdp_param_info(
+        model, set(fqn for v in param_to_fqns.values() for fqn in v)
```

Comment:
Can you explain why there is the second `for fqn in v` and also generally why we need this `dedup_shared_fqns` argument to `_get_fqn_to_fsdp_param_info`? (See my comment below at the function as well.)

```
set(fqn for v in param_to_fqns.values() for fqn in v)
```

Reply:
For `use_orig_params == False` case, there may be more than 1 FQN for a param (which is a flat param) and we need to unflatten the list. Hence the second `for`.